supportedBackends/spark/src/test/scala/cromwell/backend/impl/spark/SparkRuntimeAttributesSpec.scala

```diff
@@ -0,0 +1,131 @@
+      assertSparkRuntimeAttributesFailedCreation(runtimeAttributes, "Expecting numberOfExecutors runtime attribute to be an Integer")
+    }
+
+    "return an instance of itself when tries to validate a valid failOnStderr entry" in {
+      val expectedRuntimeAttributes = staticDefaults.copy(failOnStderr = true)
+      val runtimeAttributes = createRuntimeAttributes(HelloWorld, """runtime { failOnStderr: "true" %s: "%s" }""").head
+      val shouldBeIgnored = workflowOptionsWithDefaultRA(Map(FailOnStderrKey -> JsBoolean(false)))
+      assertSparkRuntimeAttributesSuccessfulCreation(runtimeAttributes, shouldBeIgnored, expectedRuntimeAttributes)
+    }
+
+    "throw an exception when tries to validate an invalid failOnStderr entry" in {
+      val runtimeAttributes = createRuntimeAttributes(HelloWorld, """runtime { failOnStderr: "yes" %s: "%s" }""").head
+      assertSparkRuntimeAttributesFailedCreation(runtimeAttributes, "Expecting failOnStderr runtime attribute to be a Boolean or a String with values of 'true' or 'false'")
+    }
+
+    "use workflow options as default if failOnStdErr key is missing" in {
+      val runtimeAttributes = createRuntimeAttributes(HelloWorld, """runtime { %s: "%s" }""").head
+      val workflowOptions = workflowOptionsWithDefaultRA(Map(FailOnStderrKey -> JsBoolean(true)))
+      assertSparkRuntimeAttributesSuccessfulCreation(runtimeAttributes, workflowOptions, staticDefaults.copy(failOnStderr = true))
+    }
+
+  }
+
+  private def buildWorkflowDescriptor(wdl: WdlSource,
+                                      inputs: Map[String, WdlValue] = Map.empty,
+                                      options: WorkflowOptions = WorkflowOptions(JsObject(Map.empty[String, JsValue])),
+                                      runtime: String = "") = {
+    new BackendWorkflowDescriptor(
+      WorkflowId.randomId(),
+      NamespaceWithWorkflow.load(wdl.replaceAll("RUNTIME", runtime.format("appMainClass", "com.test.spark"))),
+      inputs,
+      options
+    )
+  }
+
+  private def createRuntimeAttributes(wdlSource: WdlSource, runtimeAttributes: String = "") = {
+    val workflowDescriptor = buildWorkflowDescriptor(wdlSource, runtime = runtimeAttributes)
+
+    def createLookup(call: Call): ScopedLookupFunction = {
+      val declarations = workflowDescriptor.workflowNamespace.workflow.declarations ++ call.task.declarations
+      val knownInputs = workflowDescriptor.inputs
+      WdlExpression.standardLookupFunction(knownInputs, declarations, NoFunctions)
+    }
+
+    workflowDescriptor.workflowNamespace.workflow.calls map {
+      call =>
+        val ra = call.task.runtimeAttributes.attrs mapValues { _.evaluate(createLookup(call), NoFunctions) }
+        TryUtil.sequenceMap(ra, "Runtime attributes evaluation").get
+    }
+  }
+
+  private def assertSparkRuntimeAttributesSuccessfulCreation(runtimeAttributes: Map[String, WdlValue], workflowOptions: WorkflowOptions, expectedRuntimeAttributes: SparkRuntimeAttributes): Unit = {
```

Comment:
The name is misleading, this not only asserts successful creation but also equality.